test/framework/context.go

```diff
@@ -53,7 +53,11 @@ func (f *Framework) NewTestCtx(t *testing.T) *TestCtx {
 // cleanUpFns. This can e.g. be used to create multiple namespaces in the same
 // test context.
 func (ctx *TestCtx) GetObjID() string {
```

Comment:
To be honest, I never really looked into this code but I believe that we can refactor a bit and make the interfaces a bit clearer:
* I'd move the logic about truncating the string to 63 chars to `*Framework.CreateNamespace()`  since this is really where it matters.
* I'd also not expose the `ID` field (e.g. rename to `id` ) & rename this function to `ID()`  
* Leverage [UniqueVolumeName()](https://github.com/prometheus-operator/prometheus-operator/blob/4decb4b8690cbcea2308ef8c1a63f737d7f781fa/pkg/k8sutil/k8sutil.go#L271) to generate the unique name.

Reply:
Done 👍 I didn't do it because I wanted to check with you but I think it also makes sense to rename `UniqueVolumeName`  to something like `UniqueName` since it will probably be useful in other parts of the code.